mease_lab_to_nwb/convert_syntalos/syntalosrecordinginterface.py

```diff
@@ -1,31 +1,85 @@
 """Authors: Cody Baker and Ben Dichter."""
 import numpy as np
+from tempfile import TemporaryFile
 
 from pynwb import NWBFile, TimeSeries
-from nwb_conversion_tools import IntanRecordingInterface
+from nwb_conversion_tools.baserecordingextractorinterface import BaseRecordingExtractorInterface
 from hdmf.data_utils import DataChunkIterator
 from hdmf.backends.hdf5.h5_utils import H5DataIO
 
+from .syntalosrecordingextractor import SyntalosRecordingExtractor
+
 
 def all_equal(lst: list):
     """Determine if all elements of a list are equal."""
     return len(set(lst)) == 1
 
 
-def get_block_info(shape, size):
-    """Calculate block information in binary file."""
-    assert len(shape) == 2, "The shape of the blocks in the binary file must have two dimensions!"
-    block_size = shape[1]
-    block_stop = size // block_size + 1
-    sl0 = block_size
-    sl1 = sl0 + size
-    return block_stop, sl0, sl1
+# TODO: allow buffer_mb to be specified top-down (currently defaulting)
+def write_accelerometer_data(nwbfile: NWBFile, recording, stub_test: bool = False, buffer_mb: int = 500):
+    """Add accelerometer data from a single rhd file to the NWBFile."""
+    accel_channels = np.array([ch for ch in recording._recordings[0]._recording._anas_chan if 'AUX' in ch['name']])
+    channel_conversion = [x['gain'] for x in accel_channels]
+    accel_channel_units = [x['units'] for x in accel_channels]
+    accel_channel_sampling_rate = [x['sampling_rate'] for x in accel_channels]
+
+    if not all_equal(channel_conversion):
+        raise NotImplementedError("Writing TimeSeries in NWBFiles with channel conversion factors is not supported!")
+    if not all([x['offset'] == 0 for x in accel_channels]):
+        raise NotImplementedError("Unable to support non-zero offsets for auxiliary channel data.")
+    if not all_equal(accel_channel_units):
+        raise NotImplementedError("Unequal auxiliary channel unit types are not yet supported.")
+    if not all_equal(accel_channel_sampling_rate):
+        raise NotImplementedError("Unequal auxiliary channel sampling rates are not yet supported.")
+    if not isinstance(recording._recordings[0]._recording._raw_data[accel_channels[0]['name']], np.memmap):
+        raise NotImplementedError("Writing accelerometer data for non-memory maps is not supported!")
+
+    conversion = channel_conversion[0]
+    all_memmaps = [
+        [x._recording._raw_data[accel_channels[j]['name']].flatten() for j in range(accel_channels.size)]
+        for x in recording._recordings
+    ]
+    lens = [x[0].size for x in all_memmaps]
+    total_length = sum(lens)
+    all_accel_data = np.memmap(
+        filename=TemporaryFile(),
+        dtype=recording.get_dtype(),
+        mode="w+",
+        shape=(total_length, accel_channels.size)
+    )
```

Comment:
Why write this to a temporary file?

Reply:
I tried to figure out a way around this, but `np.memmap` requires a file on disc to create the memory map; it also must be read in as an append type mode "`w+`"/"`r+`" indicating the shape may not match the full size of the contents. If we use either of these on the original binary files, they become corrupted by anything we do to the total `all_accel_data` `memmap` after the fact.

I tested the temporary file approach, and can assure that these files on disk are quite small and automatically removed upon closing. The I/O writing and RAM usage of the entire process is also quite streamlined by this approach.

I'm open to other approaches to concatenating the memory maps into a single `memmap` however.

Reply:
This type of action could also, if accepted, be extended to some of the cases on `spikeextractors` where binary files return arrays instead of memory maps due to indexing.

Reply:
I'm confused what you are trying to accomplish. Why does this need to be a memmap?

Reply:
In the multi-file setup, there are multiple binary files. We can read those in as a list of memory maps, but to avoid the use of an array based data generator in conjunction with the `DataChunkIterator`, it is far simpler if the object you pass in for data is a single memory map.

So all this does is take that list of memory maps and concatenate them into a single memory map object that can be read in using the established controls for RAM specification identical to what we use in the `add_electrical_series` over in `spikeextractors`.

This approach offers much greater control over the amount of RAM that any given chunk being read in has, so for a large number of binary files (or even a small number) this is far more efficient, as we've learned from use of that method on the Yuta dataset.

Reply:
Can't you just concatenate the data into a np.array?

Reply:
We could, and yeah accelerometer data is at a slightly lower sampling rate so if the recording isn't SUPER long, loading it all as an array _might_ be OK.

Then again, this method is already working - would you want me to undo it in favor of the more RAM intensive method?

Reply:
Let's just add it as a numpy array for now. I really doubt we'd run into RAM problems with the accelerometer data, so this seems more complicated than it needs to be. If we do run into issues, we should work on a dedicated `DataChunkIterator` to deal with it.